lib/model-helper.js

```diff
@@ -103,14 +121,25 @@ var modelHelper = module.exports = {
     });
 
     out[name] = {
-      id: name,
-      properties: properties,
-      required: required
+      id: name
     };
-    if (def.settings && typeof def.settings.additionalProperties != 'undefined') {
-      out[name].additionalProperties = def.settings.additionalProperties;
+
+    if (def.settings){
+      var strict = def.settings.strict;
+      var additionalProperties = def.settings.additionalProperties;
+      var notAllowAdditionalProperties = strict || (additionalProperties !== true);
+      if (notAllowAdditionalProperties){
+        out[name].additionalProperties = !notAllowAdditionalProperties;
```

Comment:
Why not just `false` instead of `!notAllowAdditionalProperties`?  Or remove the `if`, and always set to `!notAllowAdditionalProperties` instead